common_code/inspiral/CE_orbit.f90

```diff
@@ -29,10 +29,19 @@ module CE_orbit
       use const_def
       use math_lib
 
+      use dop853_module, wp => dop853_wp
+      use iso_fortran_env, only: output_unit
+
+      
+
+
 !      use binary_evolve, only: eval_rlobe
 
       implicit none
 
+      integer :: id_dop ! these variables cannot be arguments of this function
+      integer :: ierr_dop
```

Comment:
Please update the comment: You need those variables to be transported between two of our functions, where one of them is an argument for the integrator.
I think, there should be a better way than using global variables here. Did you tried to make it at least private in the CE_orbit module? Another idea could be to add the information to the integrated vector with a derivative with respect to the independent variable to be 0, hence not changing (Drawback: it may lowers the performance of the integrator if it doesn't recognize a 0 change.).